Serialization Bug

#5 is not completed yet, but it had to be merged because of the breaking changes in the model and services. There is a pending bug that will be solved as a part of this issue.

Concretely:

```bash
[error] (run-main-0) io.grpc.StatusRuntimeException: CANCELLED: Failed to stream message
[error] io.grpc.StatusRuntimeException: CANCELLED: Failed to stream message
[error] 	at io.grpc.Status.asRuntimeException(Status.java:526)
[error] 	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:433)
[error] 	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:422)
[error] 	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:61)
[error] 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:504)
[error] 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$600(ClientCallImpl.java:425)
[error] 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:536)
[error] 	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
[error] 	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:102)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] Caused by: com.google.protobuf.InvalidProtocolBufferException: While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.
[error] 	at com.google.protobuf.InvalidProtocolBufferException.truncatedMessage(InvalidProtocolBufferException.java:82)
[error] 	at com.google.protobuf.CodedInputStream$ArrayDecoder.readRawBytes(CodedInputStream.java:1186)
[error] 	at com.google.protobuf.CodedInputStream$ArrayDecoder.readByteArray(CodedInputStream.java:846)
[error] 	at pbdirect.package$PBWriterOps.toPB(package.scala:16)
[error] 	at freestyle.rpc.internal.service.encoders$$anon$1.stream(service
```

--- a/bench/src/main/scala/http/ClientApp.scala
+++ b/bench/src/main/scala/http/ClientApp.scala
@@ -59,7 +59,7 @@ object ClientApp {
 
     val result: PersonAggregation = aggregation.unsafePerformSync
 
-    println(result)
+    println(s"Result = $result")
 
   }
 

--- a/bench/src/main/scala/rpc/ClientApp.scala
+++ b/bench/src/main/scala/rpc/ClientApp.scala
@@ -53,9 +53,6 @@ trait SampleClient {
 
 object ClientProgram {
 
-  //TODO Proper PersonAggregation is defined at metrifier.implicits#PersonAggregation
-
-
   def clientProgram[M[_]](implicit APP: SampleClient[M]): FreeS[M, PersonAggregation] = {
     for {
       personList <- APP.listPersons

--- a/bench/src/main/scala/rpc/handlers/PersonClientHandler.scala
+++ b/bench/src/main/scala/rpc/handlers/PersonClientHandler.scala
@@ -20,7 +20,7 @@ class SampleClientHandler[F[_]: Monad](
   override def listPersons: F[PersonList] =
     M.handleErrorWith {
       logger.info(s"*** listPersons ***")
-      client.listPersons(true)
+      client.listPersons("foo")
         .map { personList: PersonList =>
           logger.info(s"Found '${personList.count}' persons: ${personList.persons}")
           personList

--- a/frees-rpc/src/main/scala/handlers/PersonServiceHandler.scala
+++ b/frees-rpc/src/main/scala/handlers/PersonServiceHandler.scala
@@ -10,7 +10,7 @@ class PersonServiceHandler[F[_]](implicit C: Capture[F]) extends PersonService.H
 
   val logger: Logger = Logger[this.type]
 
-  override def listPersons(b: Boolean): F[PersonList] = C.capture {
+  override def listPersons(b: String): F[PersonList] = C.capture {
     logger.info(s" ### listPersons ### ")
     services.listPersons
   }

--- a/frees-rpc/src/main/scala/service.scala
+++ b/frees-rpc/src/main/scala/service.scala
@@ -29,7 +29,7 @@ object protocols {
   @service
   trait PersonService {
 
-    @rpc def listPersons(b: Boolean): FS[PersonList]
+    @rpc def listPersons(b: String): FS[PersonList]
 
     @rpc def getPerson(id: String): FS[Person]
 

